### PR TITLE
feat: custom writer for progressbar and spinner printers

### LINKED ISSUE #261: Add ability to set custom io.writter for each of the different output levels
This would allow for errors to be written to standard error or for debug logs to be written to a file.

### PR DESCRIPTION
**Pull request by Brookke**
### Description
Adds a custom writer to the progress bar and spinner components.

ProgressBar and Spinner

- [ ] Bug Fix
- [x] New Feature
- [ ] Documentation
- [ ] Other

### Related Issue
Fixes #261 

### To-Do Checklist
- [ ] I tested my changes
- [x] I have commented every method that I created/changed
- [ ] I updated the examples to fit with my changes
- [x] I have added tests for my newly created methods

### CODE REVIEW

File: docs/docs/printer/progressbar.md [hunk 1]
```diff
@@ -23,12 +22,12 @@ progressbar.Increment()
 
 ### Functions
 
-|Function|Description|
-|--------|-----------|
-|[Add(count int)](https://pkg.go.dev/github.com/pterm/pterm#TemplatePrinter.Add)|Add `count` to current value.|
-|[GetElapsedTime()](https://pkg.go.dev/github.com/pterm/pterm#TemplatePrinter.GetElapsedTime)|GetElapsedTime returns the elapsed time, since the ProgressbarPrinter was started.|
-|[Increment()](https://pkg.go.dev/github.com/pterm/pterm#TemplatePrinter.Increment)|Increment current value by one.|
-|[UpdateTitle(title string)](https://pkg.go.dev/github.com/pterm/pterm#TemplatePrinter.UpdateTitle)|Update the progressbar's title.|
```
Comment (Brookke) [inline docs/docs/printer/progressbar.md:31]: These links seemed wrong so I updated them too :)

**Comment by lowdewijk** [comment]
Exactly what I need! Was actually considering doing this myself. 👍

**Comment by MarvinJWendt** [comment]
Hi @Brookke, thanks for the PR! Could you maybe add some tests to ensure, the output is actually written to the given writer? You can do that with testza: https://github.com/MarvinJWendt/testza#capturestderr

You could use the writer in the callback function to write to and assert the output contains a word you put into the progressbar / spinner title. Thanks!

**Review by MarvinJWendt — CHANGES_REQUESTED** [review]
Some more unit tests, that capture the output and ensure that a specific writer is used, are missing. If you need help with testza, just ping me here or on Discord :)

**Comment by Brookke** [comment]
@MarvinJWendt I have added some tests for the spinner printer, and I will do the same for the progress bar soon. However I had to add time.sleeps in order to wait for the go routines to run, which isn't great. If you have any other suggestions do let me know

**Comment by Brookke** [comment]
Additionally it looks like the tests fail on ubuntu, I will investigate why

**Comment by MarvinJWendt** [comment]
> @MarvinJWendt I have added some tests for the spinner printer, and I will do the same for the progress bar soon. However I had to add time.sleeps in order to wait for the go routines to run, which isn't great. If you have any other suggestions do let me know

The CI-System takes ages to complete already, so I don't think the second delay is a problem, as I also can't think of an easy solution.

**Comment by Brookke** [comment]
@MarvinJWendt great thanks, I'll do the same for the progress back early next week, currently on holiday :)

**Review by MarvinJWendt — APPROVED** [review]
I hope you enjoyed your holidays! The PR looks good to me!

**Pull request merged**